<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>GO gene set enrichment analysis</title>
  <meta name="description" content="GO gene set enrichment analysis" />
  <meta name="generator" content="bookdown 0.44 and GitBook 2.6.7" />

  <meta property="og:title" content="GO gene set enrichment analysis" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="GO gene set enrichment analysis" />
  
  
  

<meta name="author" content="Mason Matich" />


<meta name="date" content="2025-10-20" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="full-lotr-1-ko-vs-wt-lysate.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<script src="libs/jszip-1.13.6/jszip.min.js"></script>
<script src="libs/pdfmake-1.13.6/pdfmake.js"></script>
<script src="libs/pdfmake-1.13.6/vfs_fonts.js"></script>
<link href="libs/dt-ext-buttons-1.13.6/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="libs/dt-ext-buttons-1.13.6/js/dataTables.buttons.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.6/js/buttons.html5.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.6/js/buttons.colVis.min.js"></script>
<script src="libs/dt-ext-buttons-1.13.6/js/buttons.print.min.js"></script>
<link href="libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">bulk RNA-seq Analysis</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html"><i class="fa fa-check"></i>Setup</a>
<ul>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#function-called-for-analysis"><i class="fa fa-check"></i>Function called for Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#compileontologyreport.r"><i class="fa fa-check"></i>compileOntologyReport.R</a></li>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#translatebioids.r"><i class="fa fa-check"></i>translateBioIDs.R</a></li>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#callclusterprofilerfunc.r"><i class="fa fa-check"></i>callClusterProfilerFunc.R</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#read-in-data"><i class="fa fa-check"></i>Read in Data</a></li>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#lotr-1-mutants"><i class="fa fa-check"></i>LOTR-1 Mutants</a></li>
<li class="chapter" data-level="" data-path="setup.html"><a href="setup.html#get-deseq2-results"><i class="fa fa-check"></i>Get DESeq2 Results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="full-lotr-1-ko-vs-wt-lysate.html"><a href="full-lotr-1-ko-vs-wt-lysate.html"><i class="fa fa-check"></i>Full LOTR-1 KO vs WT Lysate</a>
<ul>
<li class="chapter" data-level="" data-path="full-lotr-1-ko-vs-wt-lysate.html"><a href="full-lotr-1-ko-vs-wt-lysate.html#dysregulated"><i class="fa fa-check"></i>Dysregulated</a>
<ul>
<li><a href="full-lotr-1-ko-vs-wt-lysate.html#section-1" id="toc-section-1"></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="full-lotr-1-ko-vs-wt-lysate.html"><a href="full-lotr-1-ko-vs-wt-lysate.html#upregulated"><i class="fa fa-check"></i>Upregulated</a></li>
<li class="chapter" data-level="" data-path="full-lotr-1-ko-vs-wt-lysate.html"><a href="full-lotr-1-ko-vs-wt-lysate.html#downregulated"><i class="fa fa-check"></i>Downregulated</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">GO gene set enrichment analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">GO gene set enrichment analysis</h1>
<p class="author"><em>Mason Matich</em></p>
<p class="date"><em>2025-10-20</em></p>
</div>
<div id="setup" class="section level1 title hasAnchor">
<h1>Setup<a href="setup.html#setup" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Many of the figures and analysis methodology were based off of this tutorial from NYU.
<a href="https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/" class="uri">https://learn.gencore.bio.nyu.edu/rna-seq-analysis/gene-set-enrichment-analysis/</a></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="setup.html#cb1-1" tabindex="-1"></a><span class="do">## Library Preparation</span></span>
<span id="cb1-2"><a href="setup.html#cb1-2" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-3"><a href="setup.html#cb1-3" tabindex="-1"></a><span class="fu">library</span>(ReactomePA)</span>
<span id="cb1-4"><a href="setup.html#cb1-4" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-5"><a href="setup.html#cb1-5" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-6"><a href="setup.html#cb1-6" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-7"><a href="setup.html#cb1-7" tabindex="-1"></a><span class="fu">library</span>(presto)</span>
<span id="cb1-8"><a href="setup.html#cb1-8" tabindex="-1"></a><span class="fu">library</span>(ashr)</span>
<span id="cb1-9"><a href="setup.html#cb1-9" tabindex="-1"></a><span class="fu">library</span>(GOSemSim)</span>
<span id="cb1-10"><a href="setup.html#cb1-10" tabindex="-1"></a><span class="fu">library</span>(org.Ce.eg.db)</span>
<span id="cb1-11"><a href="setup.html#cb1-11" tabindex="-1"></a><span class="fu">library</span>(bookdown)</span>
<span id="cb1-12"><a href="setup.html#cb1-12" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb1-13"><a href="setup.html#cb1-13" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-14"><a href="setup.html#cb1-14" tabindex="-1"></a><span class="fu">library</span>(htmltools)</span>
<span id="cb1-15"><a href="setup.html#cb1-15" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="setup.html#cb2-1" tabindex="-1"></a><span class="co"># Call functions</span></span>
<span id="cb2-2"><a href="setup.html#cb2-2" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;~/Documents/research_labs/mdibl_fall_2025/updike_lotr1_go_analysis/scripts/translateBioIDs.R&quot;</span>)</span>
<span id="cb2-3"><a href="setup.html#cb2-3" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;~/Documents/research_labs/mdibl_fall_2025/updike_lotr1_go_analysis/scripts/callClusterProfilerFunc.R&quot;</span>)</span>
<span id="cb2-4"><a href="setup.html#cb2-4" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;~/Documents/research_labs/mdibl_fall_2025/updike_lotr1_go_analysis/scripts/compileOntologyReport.R&quot;</span>)</span>
<span id="cb2-5"><a href="setup.html#cb2-5" tabindex="-1"></a><span class="fu">source</span>(<span class="st">&quot;~/Documents/research_labs/mdibl_fall_2025/updike_lotr1_go_analysis/scripts/emptyTableMessage.R&quot;</span>)</span></code></pre></div>
<div id="function-called-for-analysis" class="section level2 tabset tabset-pills hasAnchor">
<h2>Function called for Analysis<a href="setup.html#function-called-for-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="compileontologyreport.r" class="section level3 hasAnchor">
<h3>compileOntologyReport.R<a href="setup.html#compileontologyreport.r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="setup.html#cb3-1" tabindex="-1"></a><span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #  </span></span>
<span id="cb3-2"><a href="setup.html#cb3-2" tabindex="-1"></a><span class="co"># #         Calls translateBioIDs.R and geneOntologyAnalysis.R        # # </span></span>
<span id="cb3-3"><a href="setup.html#cb3-3" tabindex="-1"></a><span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb3-4"><a href="setup.html#cb3-4" tabindex="-1"></a> </span>
<span id="cb3-5"><a href="setup.html#cb3-5" tabindex="-1"></a>compileOntologyReport <span class="ot">&lt;-</span> <span class="cf">function</span>(result.obj, strain, fraction, l2fc_filter, padj_filter){ </span>
<span id="cb3-6"><a href="setup.html#cb3-6" tabindex="-1"></a>   </span>
<span id="cb3-7"><a href="setup.html#cb3-7" tabindex="-1"></a>  <span class="co"># Create comment structure for bookdown render </span></span>
<span id="cb3-8"><a href="setup.html#cb3-8" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;# %s %s {.title .tabset}</span><span class="sc">\n\n</span><span class="st">&quot;</span>, strain, fraction)) </span>
<span id="cb3-9"><a href="setup.html#cb3-9" tabindex="-1"></a>  <span class="co">#cat(paste0(&quot;# &quot;, strain,&quot; &quot;, fraction, &quot; {.title .tabset}\n\n&quot;)) </span></span>
<span id="cb3-10"><a href="setup.html#cb3-10" tabindex="-1"></a> </span>
<span id="cb3-11"><a href="setup.html#cb3-11" tabindex="-1"></a>  <span class="co"># Show exact call which invoked function </span></span>
<span id="cb3-12"><a href="setup.html#cb3-12" tabindex="-1"></a>  call_txt <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">deparse</span>(<span class="fu">match.call</span>()), <span class="at">collapse =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-13"><a href="setup.html#cb3-13" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;**Inputs for function call:**</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-14"><a href="setup.html#cb3-14" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;```r</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-15"><a href="setup.html#cb3-15" tabindex="-1"></a>  <span class="fu">cat</span>(call_txt, <span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-16"><a href="setup.html#cb3-16" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;```</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-17"><a href="setup.html#cb3-17" tabindex="-1"></a>   </span>
<span id="cb3-18"><a href="setup.html#cb3-18" tabindex="-1"></a>  <span class="co"># Translate Wormbase IDs to ENTREZID </span></span>
<span id="cb3-19"><a href="setup.html#cb3-19" tabindex="-1"></a>  result.obj.list <span class="ot">&lt;-</span> <span class="fu">translateBioIDs</span>( </span>
<span id="cb3-20"><a href="setup.html#cb3-20" tabindex="-1"></a>    <span class="at">DESeqResults.obj =</span> result.obj, </span>
<span id="cb3-21"><a href="setup.html#cb3-21" tabindex="-1"></a>    <span class="at">bioID =</span> <span class="st">&quot;WORMBASE&quot;</span>, </span>
<span id="cb3-22"><a href="setup.html#cb3-22" tabindex="-1"></a>    <span class="at">l2fc_filter =</span> l2fc_filter, </span>
<span id="cb3-23"><a href="setup.html#cb3-23" tabindex="-1"></a>    <span class="at">padj_filter =</span> padj_filter </span>
<span id="cb3-24"><a href="setup.html#cb3-24" tabindex="-1"></a>  ) </span>
<span id="cb3-25"><a href="setup.html#cb3-25" tabindex="-1"></a>   </span>
<span id="cb3-26"><a href="setup.html#cb3-26" tabindex="-1"></a>  <span class="co"># Run gruopGO(), enrichGO(), gseGO(), and goplot() </span></span>
<span id="cb3-27"><a href="setup.html#cb3-27" tabindex="-1"></a>  go.result.list <span class="ot">&lt;-</span> <span class="fu">callClusterProfilerFunc</span>( </span>
<span id="cb3-28"><a href="setup.html#cb3-28" tabindex="-1"></a>    <span class="at">result.obj.list =</span> result.obj.list, </span>
<span id="cb3-29"><a href="setup.html#cb3-29" tabindex="-1"></a>    <span class="at">OrgDb =</span> org.Ce.eg.db) </span>
<span id="cb3-30"><a href="setup.html#cb3-30" tabindex="-1"></a>   </span>
<span id="cb3-31"><a href="setup.html#cb3-31" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb3-32"><a href="setup.html#cb3-32" tabindex="-1"></a>  <span class="co"># #     Print text and tables to console for bookdown rendering     # # </span></span>
<span id="cb3-33"><a href="setup.html#cb3-33" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb3-34"><a href="setup.html#cb3-34" tabindex="-1"></a>   </span>
<span id="cb3-35"><a href="setup.html#cb3-35" tabindex="-1"></a>  knitDataTable <span class="ot">&lt;-</span> <span class="cf">function</span>(df, tableName, fileExtension){ </span>
<span id="cb3-36"><a href="setup.html#cb3-36" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">nrow</span>(<span class="fu">data.frame</span>(df[[<span class="dv">1</span>]])) <span class="sc">!=</span> <span class="dv">0</span>){ </span>
<span id="cb3-37"><a href="setup.html#cb3-37" tabindex="-1"></a>      table <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">datatable</span>(<span class="fu">data.frame</span>(df[[<span class="dv">1</span>]]), </span>
<span id="cb3-38"><a href="setup.html#cb3-38" tabindex="-1"></a>                            <span class="at">extensions =</span> <span class="st">&#39;Buttons&#39;</span>, </span>
<span id="cb3-39"><a href="setup.html#cb3-39" tabindex="-1"></a>                            <span class="at">rownames =</span> <span class="cn">FALSE</span>, </span>
<span id="cb3-40"><a href="setup.html#cb3-40" tabindex="-1"></a>                            <span class="at">caption =</span> tableName, </span>
<span id="cb3-41"><a href="setup.html#cb3-41" tabindex="-1"></a>                            <span class="at">options =</span> <span class="fu">list</span>( </span>
<span id="cb3-42"><a href="setup.html#cb3-42" tabindex="-1"></a>                              <span class="at">dom =</span> <span class="st">&quot;Bfrtip&quot;</span>, </span>
<span id="cb3-43"><a href="setup.html#cb3-43" tabindex="-1"></a>                              <span class="at">buttons =</span> <span class="fu">list</span>(<span class="st">&#39;copy&#39;</span>, </span>
<span id="cb3-44"><a href="setup.html#cb3-44" tabindex="-1"></a>                                 <span class="fu">list</span>(<span class="at">extend =</span> <span class="st">&#39;csv&#39;</span>, <span class="at">filename =</span> <span class="fu">sprintf</span>(<span class="st">&quot;%s_%s_%s&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="fu">tolower</span>(strain)), <span class="fu">tolower</span>(fraction), fileExtension)), </span>
<span id="cb3-45"><a href="setup.html#cb3-45" tabindex="-1"></a>                                 <span class="fu">list</span>(<span class="at">extend =</span> <span class="st">&#39;excel&#39;</span>, <span class="at">filename =</span> <span class="fu">sprintf</span>(<span class="st">&quot;%s_%s_%s&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="fu">tolower</span>(strain)), <span class="fu">tolower</span>(fraction), fileExtension)) </span>
<span id="cb3-46"><a href="setup.html#cb3-46" tabindex="-1"></a>                              ), </span>
<span id="cb3-47"><a href="setup.html#cb3-47" tabindex="-1"></a>                              <span class="at">pageLength =</span> <span class="dv">5</span>, </span>
<span id="cb3-48"><a href="setup.html#cb3-48" tabindex="-1"></a>                              <span class="at">scrollX =</span> <span class="cn">TRUE</span>))) </span>
<span id="cb3-49"><a href="setup.html#cb3-49" tabindex="-1"></a>      <span class="fu">print</span>(htmltools<span class="sc">::</span><span class="fu">tagList</span>(table[[<span class="dv">1</span>]])) </span>
<span id="cb3-50"><a href="setup.html#cb3-50" tabindex="-1"></a>      <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-51"><a href="setup.html#cb3-51" tabindex="-1"></a>      <span class="fu">write.csv</span>(df[[<span class="dv">1</span>]], <span class="at">file =</span> <span class="fu">sprintf</span>(<span class="st">&quot;./results/%s_%s_%s.csv&quot;</span>, <span class="fu">gsub</span>(<span class="st">&quot; &quot;</span>, <span class="st">&quot;_&quot;</span>, <span class="fu">tolower</span>(strain)), <span class="fu">tolower</span>(fraction), fileExtension)) </span>
<span id="cb3-52"><a href="setup.html#cb3-52" tabindex="-1"></a>    }<span class="cf">else</span>{<span class="fu">emptyTableMessage</span>(tableName)} </span>
<span id="cb3-53"><a href="setup.html#cb3-53" tabindex="-1"></a>     </span>
<span id="cb3-54"><a href="setup.html#cb3-54" tabindex="-1"></a>    <span class="co"># clean up environment </span></span>
<span id="cb3-55"><a href="setup.html#cb3-55" tabindex="-1"></a>    <span class="fu">rm</span>(table) </span>
<span id="cb3-56"><a href="setup.html#cb3-56" tabindex="-1"></a>  } </span>
<span id="cb3-57"><a href="setup.html#cb3-57" tabindex="-1"></a>   </span>
<span id="cb3-58"><a href="setup.html#cb3-58" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb3-59"><a href="setup.html#cb3-59" tabindex="-1"></a>  <span class="co"># # #          Dysregulated Reactome Pathway GSE, GO GSE          # # #  </span></span>
<span id="cb3-60"><a href="setup.html#cb3-60" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">## Dysregulated</span><span class="sc">\n\n</span><span class="st">&quot;</span>)) </span>
<span id="cb3-61"><a href="setup.html#cb3-61" tabindex="-1"></a>   </span>
<span id="cb3-62"><a href="setup.html#cb3-62" tabindex="-1"></a>  <span class="co"># Reactome Pathway GSE </span></span>
<span id="cb3-63"><a href="setup.html#cb3-63" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;**Reactome Pathway GSEA**&quot;</span>) </span>
<span id="cb3-64"><a href="setup.html#cb3-64" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-65"><a href="setup.html#cb3-65" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>react_gse, </span>
<span id="cb3-66"><a href="setup.html#cb3-66" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; Reactome Gene Set Enrichment Analysis&quot;</span>, </span>
<span id="cb3-67"><a href="setup.html#cb3-67" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;Reactome_gsea&quot;</span> </span>
<span id="cb3-68"><a href="setup.html#cb3-68" tabindex="-1"></a>  ) </span>
<span id="cb3-69"><a href="setup.html#cb3-69" tabindex="-1"></a>   </span>
<span id="cb3-70"><a href="setup.html#cb3-70" tabindex="-1"></a>  <span class="co"># # # GSEA plot condiitonal rendering of RP GSEA plots </span></span>
<span id="cb3-71"><a href="setup.html#cb3-71" tabindex="-1"></a>  <span class="co"># Tabset for Reactome GSEA plots </span></span>
<span id="cb3-72"><a href="setup.html#cb3-72" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;#### {.tabset .tabset-dropdown}</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-73"><a href="setup.html#cb3-73" tabindex="-1"></a> </span>
<span id="cb3-74"><a href="setup.html#cb3-74" tabindex="-1"></a>  <span class="co"># Print GSEA plot for each react pathway set identified  </span></span>
<span id="cb3-75"><a href="setup.html#cb3-75" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(<span class="fu">data.frame</span>(go.result.list<span class="sc">$</span>react_gse[[<span class="dv">1</span>]]))) { </span>
<span id="cb3-76"><a href="setup.html#cb3-76" tabindex="-1"></a>    <span class="cf">if</span> (i <span class="sc">&gt;</span> <span class="dv">30</span>) {<span class="cf">break</span>} <span class="co"># Dont print over 30 plots </span></span>
<span id="cb3-77"><a href="setup.html#cb3-77" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;##### %s </span><span class="sc">\n\n</span><span class="st">&quot;</span>, go.result.list<span class="sc">$</span>react_gse[[<span class="dv">1</span>]]<span class="sc">$</span>Description[i])) </span>
<span id="cb3-78"><a href="setup.html#cb3-78" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">gseaplot</span>( </span>
<span id="cb3-79"><a href="setup.html#cb3-79" tabindex="-1"></a>      go.result.list<span class="sc">$</span>react_gse[[<span class="dv">1</span>]],  </span>
<span id="cb3-80"><a href="setup.html#cb3-80" tabindex="-1"></a>      <span class="at">by =</span> <span class="st">&quot;all&quot;</span>,  </span>
<span id="cb3-81"><a href="setup.html#cb3-81" tabindex="-1"></a>      <span class="at">title =</span> go.result.list<span class="sc">$</span>react_gse[[<span class="dv">1</span>]]<span class="sc">$</span>Description[i],  </span>
<span id="cb3-82"><a href="setup.html#cb3-82" tabindex="-1"></a>      <span class="at">geneSetID =</span> go.result.list<span class="sc">$</span>react_gse[[<span class="dv">1</span>]]<span class="sc">$</span>ID[i])) </span>
<span id="cb3-83"><a href="setup.html#cb3-83" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-84"><a href="setup.html#cb3-84" tabindex="-1"></a>  } </span>
<span id="cb3-85"><a href="setup.html#cb3-85" tabindex="-1"></a>   </span>
<span id="cb3-86"><a href="setup.html#cb3-86" tabindex="-1"></a>  <span class="co"># GO GSE </span></span>
<span id="cb3-87"><a href="setup.html#cb3-87" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;### </span><span class="sc">\n\n</span><span class="st">&quot;</span>) <span class="co"># Break out of previous tabset </span></span>
<span id="cb3-88"><a href="setup.html#cb3-88" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;**GO GSEA**&quot;</span>) </span>
<span id="cb3-89"><a href="setup.html#cb3-89" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-90"><a href="setup.html#cb3-90" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>go_gse, </span>
<span id="cb3-91"><a href="setup.html#cb3-91" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; GSE Of All Differentially Expressed Genes&quot;</span>, </span>
<span id="cb3-92"><a href="setup.html#cb3-92" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;GO_gsea&quot;</span> </span>
<span id="cb3-93"><a href="setup.html#cb3-93" tabindex="-1"></a>  ) </span>
<span id="cb3-94"><a href="setup.html#cb3-94" tabindex="-1"></a>   </span>
<span id="cb3-95"><a href="setup.html#cb3-95" tabindex="-1"></a>  <span class="co"># # # GSEA plot condiitonal rendering </span></span>
<span id="cb3-96"><a href="setup.html#cb3-96" tabindex="-1"></a>  <span class="co"># Tabset for GSEA plots </span></span>
<span id="cb3-97"><a href="setup.html#cb3-97" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;#### {.tabset .tabset-dropdown}</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-98"><a href="setup.html#cb3-98" tabindex="-1"></a> </span>
<span id="cb3-99"><a href="setup.html#cb3-99" tabindex="-1"></a>  <span class="co"># Print GSEA plot for each set identified  </span></span>
<span id="cb3-100"><a href="setup.html#cb3-100" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(<span class="fu">data.frame</span>(go.result.list<span class="sc">$</span>go_gse[[<span class="dv">1</span>]]))) { </span>
<span id="cb3-101"><a href="setup.html#cb3-101" tabindex="-1"></a>    <span class="cf">if</span> (i <span class="sc">&gt;</span> <span class="dv">30</span>) {<span class="cf">break</span>} <span class="co"># Dont print over 30 plots </span></span>
<span id="cb3-102"><a href="setup.html#cb3-102" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;##### %s </span><span class="sc">\n\n</span><span class="st">&quot;</span>, go.result.list<span class="sc">$</span>go_gse[[<span class="dv">1</span>]]<span class="sc">$</span>Description[i])) </span>
<span id="cb3-103"><a href="setup.html#cb3-103" tabindex="-1"></a>    <span class="fu">print</span>(<span class="fu">gseaplot</span>( </span>
<span id="cb3-104"><a href="setup.html#cb3-104" tabindex="-1"></a>      go.result.list<span class="sc">$</span>go_gse[[<span class="dv">1</span>]],  </span>
<span id="cb3-105"><a href="setup.html#cb3-105" tabindex="-1"></a>      <span class="at">by =</span> <span class="st">&quot;all&quot;</span>,  </span>
<span id="cb3-106"><a href="setup.html#cb3-106" tabindex="-1"></a>      <span class="at">title =</span> go.result.list<span class="sc">$</span>go_gse[[<span class="dv">1</span>]]<span class="sc">$</span>Description[i],  </span>
<span id="cb3-107"><a href="setup.html#cb3-107" tabindex="-1"></a>      <span class="at">geneSetID =</span> go.result.list<span class="sc">$</span>go_gse[[<span class="dv">1</span>]]<span class="sc">$</span>ID[i])) </span>
<span id="cb3-108"><a href="setup.html#cb3-108" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb3-109"><a href="setup.html#cb3-109" tabindex="-1"></a>  } </span>
<span id="cb3-110"><a href="setup.html#cb3-110" tabindex="-1"></a> </span>
<span id="cb3-111"><a href="setup.html#cb3-111" tabindex="-1"></a>   </span>
<span id="cb3-112"><a href="setup.html#cb3-112" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb3-113"><a href="setup.html#cb3-113" tabindex="-1"></a>  <span class="co"># # #    Upregulated GO classification, GO over-representation, RP over-representation    # # # </span></span>
<span id="cb3-114"><a href="setup.html#cb3-114" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">## Upregulated</span><span class="sc">\n\n</span><span class="st">&quot;</span>)) </span>
<span id="cb3-115"><a href="setup.html#cb3-115" tabindex="-1"></a>   </span>
<span id="cb3-116"><a href="setup.html#cb3-116" tabindex="-1"></a>  <span class="co"># GO classification </span></span>
<span id="cb3-117"><a href="setup.html#cb3-117" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-118"><a href="setup.html#cb3-118" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>go_class_upreg, </span>
<span id="cb3-119"><a href="setup.html#cb3-119" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; GO Classification of Upregulated Genes&quot;</span>, </span>
<span id="cb3-120"><a href="setup.html#cb3-120" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;GO_classification_upreg&quot;</span> </span>
<span id="cb3-121"><a href="setup.html#cb3-121" tabindex="-1"></a>  ) </span>
<span id="cb3-122"><a href="setup.html#cb3-122" tabindex="-1"></a>   </span>
<span id="cb3-123"><a href="setup.html#cb3-123" tabindex="-1"></a>  <span class="co"># GO over-representation </span></span>
<span id="cb3-124"><a href="setup.html#cb3-124" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-125"><a href="setup.html#cb3-125" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>go_overrep_upreg, </span>
<span id="cb3-126"><a href="setup.html#cb3-126" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; GO Over-representation Analysis of Upregulated Genes&quot;</span>, </span>
<span id="cb3-127"><a href="setup.html#cb3-127" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;GO_overrepresented_upreg&quot;</span> </span>
<span id="cb3-128"><a href="setup.html#cb3-128" tabindex="-1"></a>  ) </span>
<span id="cb3-129"><a href="setup.html#cb3-129" tabindex="-1"></a>   </span>
<span id="cb3-130"><a href="setup.html#cb3-130" tabindex="-1"></a>  <span class="co"># RP over-representation </span></span>
<span id="cb3-131"><a href="setup.html#cb3-131" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-132"><a href="setup.html#cb3-132" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>react_overrep_upreg, </span>
<span id="cb3-133"><a href="setup.html#cb3-133" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; Reactome Pathway Over-representation Analysis of Upregulated Genes&quot;</span>, </span>
<span id="cb3-134"><a href="setup.html#cb3-134" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;Reactome_overrepresented_upreg&quot;</span> </span>
<span id="cb3-135"><a href="setup.html#cb3-135" tabindex="-1"></a>  ) </span>
<span id="cb3-136"><a href="setup.html#cb3-136" tabindex="-1"></a>   </span>
<span id="cb3-137"><a href="setup.html#cb3-137" tabindex="-1"></a>   </span>
<span id="cb3-138"><a href="setup.html#cb3-138" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb3-139"><a href="setup.html#cb3-139" tabindex="-1"></a>  <span class="co"># # #    Downregulated GO classification, GO over-representation, RP over-representation    # # # </span></span>
<span id="cb3-140"><a href="setup.html#cb3-140" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">## Downregulated</span><span class="sc">\n\n</span><span class="st">&quot;</span>)) </span>
<span id="cb3-141"><a href="setup.html#cb3-141" tabindex="-1"></a>   </span>
<span id="cb3-142"><a href="setup.html#cb3-142" tabindex="-1"></a>  <span class="co"># GO classification </span></span>
<span id="cb3-143"><a href="setup.html#cb3-143" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-144"><a href="setup.html#cb3-144" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>go_class_downreg, </span>
<span id="cb3-145"><a href="setup.html#cb3-145" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; GO Classification of Downregulated Genes&quot;</span>, </span>
<span id="cb3-146"><a href="setup.html#cb3-146" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;GO_classification_downreg&quot;</span> </span>
<span id="cb3-147"><a href="setup.html#cb3-147" tabindex="-1"></a>  ) </span>
<span id="cb3-148"><a href="setup.html#cb3-148" tabindex="-1"></a> </span>
<span id="cb3-149"><a href="setup.html#cb3-149" tabindex="-1"></a>  <span class="co"># GO over-representation </span></span>
<span id="cb3-150"><a href="setup.html#cb3-150" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-151"><a href="setup.html#cb3-151" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>go_overrep_downreg, </span>
<span id="cb3-152"><a href="setup.html#cb3-152" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; GO Over-representation Analysis of Downregulated Genes&quot;</span>, </span>
<span id="cb3-153"><a href="setup.html#cb3-153" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;GO_overrepresented_downreg&quot;</span> </span>
<span id="cb3-154"><a href="setup.html#cb3-154" tabindex="-1"></a>  ) </span>
<span id="cb3-155"><a href="setup.html#cb3-155" tabindex="-1"></a>   </span>
<span id="cb3-156"><a href="setup.html#cb3-156" tabindex="-1"></a>  <span class="co"># RP over-representation </span></span>
<span id="cb3-157"><a href="setup.html#cb3-157" tabindex="-1"></a>  <span class="fu">knitDataTable</span>( </span>
<span id="cb3-158"><a href="setup.html#cb3-158" tabindex="-1"></a>    <span class="at">df =</span> go.result.list<span class="sc">$</span>react_overrep_downreg, </span>
<span id="cb3-159"><a href="setup.html#cb3-159" tabindex="-1"></a>    <span class="at">tableName =</span> <span class="st">&quot; Reactome Pathway Over-representation Analysis of Downregulated Genes&quot;</span>, </span>
<span id="cb3-160"><a href="setup.html#cb3-160" tabindex="-1"></a>    <span class="at">fileExtension =</span> <span class="st">&quot;Reactome_overrepresented_downreg&quot;</span> </span>
<span id="cb3-161"><a href="setup.html#cb3-161" tabindex="-1"></a>  ) </span>
<span id="cb3-162"><a href="setup.html#cb3-162" tabindex="-1"></a> </span>
<span id="cb3-163"><a href="setup.html#cb3-163" tabindex="-1"></a>} </span></code></pre></div>
</div>
<div id="translatebioids.r" class="section level3 hasAnchor">
<h3>translateBioIDs.R<a href="setup.html#translatebioids.r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="setup.html#cb4-1" tabindex="-1"></a><span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb4-2"><a href="setup.html#cb4-2" tabindex="-1"></a><span class="co"># #                       Translating Biological IDS                      # # </span></span>
<span id="cb4-3"><a href="setup.html#cb4-3" tabindex="-1"></a><span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb4-4"><a href="setup.html#cb4-4" tabindex="-1"></a><span class="co"># # # From WormBase to ENTREZID </span></span>
<span id="cb4-5"><a href="setup.html#cb4-5" tabindex="-1"></a> </span>
<span id="cb4-6"><a href="setup.html#cb4-6" tabindex="-1"></a>translateBioIDs <span class="ot">&lt;-</span> <span class="cf">function</span> (DESeqResults.obj, bioID, <span class="at">return.success =</span> <span class="cn">TRUE</span>, l2fc_filter, padj_filter) { </span>
<span id="cb4-7"><a href="setup.html#cb4-7" tabindex="-1"></a>   </span>
<span id="cb4-8"><a href="setup.html#cb4-8" tabindex="-1"></a>  <span class="cf">if</span> (bioID <span class="sc">!=</span> <span class="st">&quot;WORMBASE&quot;</span>){ </span>
<span id="cb4-9"><a href="setup.html#cb4-9" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;⛔️ ERROR: Only WORMBASE biological ID supported as input.&quot;</span>) </span>
<span id="cb4-10"><a href="setup.html#cb4-10" tabindex="-1"></a>  } </span>
<span id="cb4-11"><a href="setup.html#cb4-11" tabindex="-1"></a>   </span>
<span id="cb4-12"><a href="setup.html#cb4-12" tabindex="-1"></a>  <span class="co"># Convert DESeq2 results object to DataFrame, and edit structure </span></span>
<span id="cb4-13"><a href="setup.html#cb4-13" tabindex="-1"></a>  gene.df <span class="ot">&lt;-</span> DESeqResults.obj <span class="sc">%&gt;%</span> </span>
<span id="cb4-14"><a href="setup.html#cb4-14" tabindex="-1"></a>    <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-15"><a href="setup.html#cb4-15" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;WORMBASE&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb4-16"><a href="setup.html#cb4-16" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(WORMBASE, log2FoldChange, pvalue, padj) </span>
<span id="cb4-17"><a href="setup.html#cb4-17" tabindex="-1"></a>   </span>
<span id="cb4-18"><a href="setup.html#cb4-18" tabindex="-1"></a>  <span class="co"># Biological ID TranslatoR </span></span>
<span id="cb4-19"><a href="setup.html#cb4-19" tabindex="-1"></a>  gene_ids <span class="ot">&lt;-</span> <span class="fu">bitr</span>(gene.df[,<span class="st">&#39;WORMBASE&#39;</span>], </span>
<span id="cb4-20"><a href="setup.html#cb4-20" tabindex="-1"></a>                   <span class="at">fromType =</span> <span class="st">&quot;WORMBASE&quot;</span>, </span>
<span id="cb4-21"><a href="setup.html#cb4-21" tabindex="-1"></a>                   <span class="at">toType =</span> <span class="st">&quot;ENTREZID&quot;</span>, </span>
<span id="cb4-22"><a href="setup.html#cb4-22" tabindex="-1"></a>                   <span class="at">OrgDb =</span> org.Ce.eg.db) </span>
<span id="cb4-23"><a href="setup.html#cb4-23" tabindex="-1"></a>   </span>
<span id="cb4-24"><a href="setup.html#cb4-24" tabindex="-1"></a>  <span class="co"># Record success of ID transfer </span></span>
<span id="cb4-25"><a href="setup.html#cb4-25" tabindex="-1"></a>  t1 <span class="ot">&lt;-</span> <span class="fu">table</span>(gene.df<span class="sc">$</span>WORMBASE <span class="sc">%in%</span> gene_ids<span class="sc">$</span>WORMBASE) </span>
<span id="cb4-26"><a href="setup.html#cb4-26" tabindex="-1"></a>   </span>
<span id="cb4-27"><a href="setup.html#cb4-27" tabindex="-1"></a>  <span class="co"># Clear duplicate genes following ID transfer </span></span>
<span id="cb4-28"><a href="setup.html#cb4-28" tabindex="-1"></a>  gene_ids <span class="ot">&lt;-</span> gene_ids <span class="sc">%&gt;%</span> </span>
<span id="cb4-29"><a href="setup.html#cb4-29" tabindex="-1"></a>    <span class="fu">distinct</span>(WORMBASE, <span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb4-30"><a href="setup.html#cb4-30" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="st">&quot;WORMBASE&quot;</span>) </span>
<span id="cb4-31"><a href="setup.html#cb4-31" tabindex="-1"></a>   </span>
<span id="cb4-32"><a href="setup.html#cb4-32" tabindex="-1"></a>  <span class="co"># Add ENTREZID to L2FC carrying DataFrame </span></span>
<span id="cb4-33"><a href="setup.html#cb4-33" tabindex="-1"></a>  gene.df <span class="ot">&lt;-</span> gene.df <span class="sc">%&gt;%</span> </span>
<span id="cb4-34"><a href="setup.html#cb4-34" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">ENTREZID =</span> gene_ids[WORMBASE, <span class="dv">1</span>]) <span class="sc">%&gt;%</span> </span>
<span id="cb4-35"><a href="setup.html#cb4-35" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(ENTREZID, log2FoldChange, pvalue, padj) <span class="sc">%&gt;%</span> </span>
<span id="cb4-36"><a href="setup.html#cb4-36" tabindex="-1"></a>    <span class="fu">subset</span>(<span class="sc">!</span><span class="fu">is.na</span>(ENTREZID)) <span class="sc">%&gt;%</span> <span class="co"># remove NA ENTREZ ID values </span></span>
<span id="cb4-37"><a href="setup.html#cb4-37" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(log2FoldChange)) <span class="co"># &lt;--- arrange genes by L2FC descending </span></span>
<span id="cb4-38"><a href="setup.html#cb4-38" tabindex="-1"></a>  <span class="fu">rownames</span>(gene.df) <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb4-39"><a href="setup.html#cb4-39" tabindex="-1"></a>   </span>
<span id="cb4-40"><a href="setup.html#cb4-40" tabindex="-1"></a>  <span class="co"># #    Subset L2FC Data    # #  </span></span>
<span id="cb4-41"><a href="setup.html#cb4-41" tabindex="-1"></a>   </span>
<span id="cb4-42"><a href="setup.html#cb4-42" tabindex="-1"></a>  <span class="co"># Subset for highly varirable genes </span></span>
<span id="cb4-43"><a href="setup.html#cb4-43" tabindex="-1"></a>  variable.genes <span class="ot">&lt;-</span> gene.df <span class="sc">%&gt;%</span> </span>
<span id="cb4-44"><a href="setup.html#cb4-44" tabindex="-1"></a>    <span class="fu">subset</span>(<span class="fu">abs</span>(log2FoldChange) <span class="sc">&gt;</span> l2fc_filter <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> padj_filter <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">isNA</span>(padj)) </span>
<span id="cb4-45"><a href="setup.html#cb4-45" tabindex="-1"></a>  <span class="fu">rownames</span>(variable.genes) <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb4-46"><a href="setup.html#cb4-46" tabindex="-1"></a>   </span>
<span id="cb4-47"><a href="setup.html#cb4-47" tabindex="-1"></a>  <span class="co"># Subset for highly upregulated genes </span></span>
<span id="cb4-48"><a href="setup.html#cb4-48" tabindex="-1"></a>  upregulated.genes <span class="ot">&lt;-</span> gene.df <span class="sc">%&gt;%</span> </span>
<span id="cb4-49"><a href="setup.html#cb4-49" tabindex="-1"></a>    <span class="fu">subset</span>(log2FoldChange <span class="sc">&gt;</span> l2fc_filter <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> padj_filter <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">isNA</span>(padj)) </span>
<span id="cb4-50"><a href="setup.html#cb4-50" tabindex="-1"></a>  <span class="fu">rownames</span>(upregulated.genes) <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb4-51"><a href="setup.html#cb4-51" tabindex="-1"></a>   </span>
<span id="cb4-52"><a href="setup.html#cb4-52" tabindex="-1"></a>  <span class="co"># Subset for highly downregulated genes </span></span>
<span id="cb4-53"><a href="setup.html#cb4-53" tabindex="-1"></a>  downregulated.genes <span class="ot">&lt;-</span> gene.df <span class="sc">%&gt;%</span> </span>
<span id="cb4-54"><a href="setup.html#cb4-54" tabindex="-1"></a>    <span class="fu">subset</span>(log2FoldChange <span class="sc">&lt;</span> <span class="sc">-</span>l2fc_filter <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> padj_filter <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">isNA</span>(padj)) </span>
<span id="cb4-55"><a href="setup.html#cb4-55" tabindex="-1"></a>  <span class="fu">rownames</span>(downregulated.genes) <span class="ot">&lt;-</span> <span class="cn">NULL</span> </span>
<span id="cb4-56"><a href="setup.html#cb4-56" tabindex="-1"></a>   </span>
<span id="cb4-57"><a href="setup.html#cb4-57" tabindex="-1"></a>  <span class="co"># Subset for all genes (universe DataFrame) </span></span>
<span id="cb4-58"><a href="setup.html#cb4-58" tabindex="-1"></a>  all.genes <span class="ot">&lt;-</span> gene.df <span class="sc">%&gt;%</span> </span>
<span id="cb4-59"><a href="setup.html#cb4-59" tabindex="-1"></a>    <span class="fu">subset</span>(<span class="sc">!</span><span class="fu">isNA</span>(padj)) <span class="sc">%&gt;%</span> </span>
<span id="cb4-60"><a href="setup.html#cb4-60" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(log2FoldChange)) </span>
<span id="cb4-61"><a href="setup.html#cb4-61" tabindex="-1"></a>   </span>
<span id="cb4-62"><a href="setup.html#cb4-62" tabindex="-1"></a>  <span class="co"># Vectorized differentially expressed genes </span></span>
<span id="cb4-63"><a href="setup.html#cb4-63" tabindex="-1"></a>  vectorized <span class="ot">&lt;-</span> all.genes[,<span class="dv">2</span>] </span>
<span id="cb4-64"><a href="setup.html#cb4-64" tabindex="-1"></a>  <span class="fu">names</span>(vectorized) <span class="ot">&lt;-</span> all.genes[,<span class="dv">1</span>] </span>
<span id="cb4-65"><a href="setup.html#cb4-65" tabindex="-1"></a>   </span>
<span id="cb4-66"><a href="setup.html#cb4-66" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb4-67"><a href="setup.html#cb4-67" tabindex="-1"></a>  <span class="co"># #                     Print Results Following Subset                    # # </span></span>
<span id="cb4-68"><a href="setup.html#cb4-68" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb4-69"><a href="setup.html#cb4-69" tabindex="-1"></a> </span>
<span id="cb4-70"><a href="setup.html#cb4-70" tabindex="-1"></a>  <span class="co"># Record transfer success </span></span>
<span id="cb4-71"><a href="setup.html#cb4-71" tabindex="-1"></a>  <span class="cf">if</span> (return.success <span class="sc">==</span> <span class="cn">TRUE</span>){ </span>
<span id="cb4-72"><a href="setup.html#cb4-72" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-73"><a href="setup.html#cb4-73" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;**Transfering WORMBASE gene IDs to ENTREZID:**</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-74"><a href="setup.html#cb4-74" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;```r</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-75"><a href="setup.html#cb4-75" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(t1[[<span class="st">&quot;FALSE&quot;</span>]],<span class="st">&quot; gene IDs were not transfered from WORMBASE to ENTREZID </span><span class="sc">\n</span><span class="st">&quot;</span>)) </span>
<span id="cb4-76"><a href="setup.html#cb4-76" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(t1[[<span class="st">&quot;TRUE&quot;</span>]],<span class="st">&quot; gene IDs were successfully transfered from WORMBASE to ENTREZID </span><span class="sc">\n\n</span><span class="st">&quot;</span>)) </span>
<span id="cb4-77"><a href="setup.html#cb4-77" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="st">&quot;```</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-78"><a href="setup.html#cb4-78" tabindex="-1"></a>  } </span>
<span id="cb4-79"><a href="setup.html#cb4-79" tabindex="-1"></a>   </span>
<span id="cb4-80"><a href="setup.html#cb4-80" tabindex="-1"></a>  <span class="co"># report number of highly variable genes used for analysis </span></span>
<span id="cb4-81"><a href="setup.html#cb4-81" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;**Number of highly variable genes identified (with | L2FC | &gt; %s and padj &lt; %s) is:**</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">toString</span>(l2fc_filter) , <span class="fu">toString</span>(padj_filter))) </span>
<span id="cb4-82"><a href="setup.html#cb4-82" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;```r</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-83"><a href="setup.html#cb4-83" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">length</span>(variable.genes[[<span class="dv">1</span>]])) </span>
<span id="cb4-84"><a href="setup.html#cb4-84" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">```</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-85"><a href="setup.html#cb4-85" tabindex="-1"></a>   </span>
<span id="cb4-86"><a href="setup.html#cb4-86" tabindex="-1"></a>  <span class="co"># report number of highly variable genes used for analysis </span></span>
<span id="cb4-87"><a href="setup.html#cb4-87" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;**Number of highly upregulated genes identified (with L2FC &gt; %s and padj &lt; %s) is:**</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">toString</span>(l2fc_filter) , <span class="fu">toString</span>(padj_filter))) </span>
<span id="cb4-88"><a href="setup.html#cb4-88" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;```r</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-89"><a href="setup.html#cb4-89" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">length</span>(upregulated.genes[[<span class="dv">1</span>]])) </span>
<span id="cb4-90"><a href="setup.html#cb4-90" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">```</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-91"><a href="setup.html#cb4-91" tabindex="-1"></a>   </span>
<span id="cb4-92"><a href="setup.html#cb4-92" tabindex="-1"></a>  <span class="co"># report number of highly variable genes used for analysis </span></span>
<span id="cb4-93"><a href="setup.html#cb4-93" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">&quot;**Number of highly downregulated genes identified (with L2FC &lt; -%s and padj &lt; %s) is:**</span><span class="sc">\n\n</span><span class="st">&quot;</span>, <span class="fu">toString</span>(l2fc_filter) , <span class="fu">toString</span>(padj_filter))) </span>
<span id="cb4-94"><a href="setup.html#cb4-94" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;```r</span><span class="sc">\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-95"><a href="setup.html#cb4-95" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">length</span>(downregulated.genes[[<span class="dv">1</span>]])) </span>
<span id="cb4-96"><a href="setup.html#cb4-96" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">```</span><span class="sc">\n\n</span><span class="st">&quot;</span>) </span>
<span id="cb4-97"><a href="setup.html#cb4-97" tabindex="-1"></a>   </span>
<span id="cb4-98"><a href="setup.html#cb4-98" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb4-99"><a href="setup.html#cb4-99" tabindex="-1"></a>  <span class="co"># #                               Return Data                             # #  </span></span>
<span id="cb4-100"><a href="setup.html#cb4-100" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb4-101"><a href="setup.html#cb4-101" tabindex="-1"></a>  <span class="co"># Save all data to list </span></span>
<span id="cb4-102"><a href="setup.html#cb4-102" tabindex="-1"></a>  gene.list <span class="ot">&lt;-</span> <span class="fu">list</span>( </span>
<span id="cb4-103"><a href="setup.html#cb4-103" tabindex="-1"></a>    variable.genes,  </span>
<span id="cb4-104"><a href="setup.html#cb4-104" tabindex="-1"></a>    upregulated.genes,  </span>
<span id="cb4-105"><a href="setup.html#cb4-105" tabindex="-1"></a>    downregulated.genes,  </span>
<span id="cb4-106"><a href="setup.html#cb4-106" tabindex="-1"></a>    all.genes,  </span>
<span id="cb4-107"><a href="setup.html#cb4-107" tabindex="-1"></a>    vectorized </span>
<span id="cb4-108"><a href="setup.html#cb4-108" tabindex="-1"></a>    ) </span>
<span id="cb4-109"><a href="setup.html#cb4-109" tabindex="-1"></a>   </span>
<span id="cb4-110"><a href="setup.html#cb4-110" tabindex="-1"></a>  <span class="fu">names</span>(gene.list) <span class="ot">&lt;-</span> <span class="fu">c</span>( </span>
<span id="cb4-111"><a href="setup.html#cb4-111" tabindex="-1"></a>    <span class="st">&quot;variable_genes&quot;</span>,  </span>
<span id="cb4-112"><a href="setup.html#cb4-112" tabindex="-1"></a>    <span class="st">&quot;upregulated_genes&quot;</span>,  </span>
<span id="cb4-113"><a href="setup.html#cb4-113" tabindex="-1"></a>    <span class="st">&quot;downregulated_genes&quot;</span>,  </span>
<span id="cb4-114"><a href="setup.html#cb4-114" tabindex="-1"></a>    <span class="st">&quot;all_genes&quot;</span>,  </span>
<span id="cb4-115"><a href="setup.html#cb4-115" tabindex="-1"></a>    <span class="st">&quot;vectorized_all_DE&quot;</span> </span>
<span id="cb4-116"><a href="setup.html#cb4-116" tabindex="-1"></a>    ) </span>
<span id="cb4-117"><a href="setup.html#cb4-117" tabindex="-1"></a>   </span>
<span id="cb4-118"><a href="setup.html#cb4-118" tabindex="-1"></a>  <span class="fu">return</span>(gene.list) </span>
<span id="cb4-119"><a href="setup.html#cb4-119" tabindex="-1"></a>} </span></code></pre></div>
</div>
<div id="callclusterprofilerfunc.r" class="section level3 hasAnchor">
<h3>callClusterProfilerFunc.R<a href="setup.html#callclusterprofilerfunc.r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="setup.html#cb5-1" tabindex="-1"></a><span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-2"><a href="setup.html#cb5-2" tabindex="-1"></a><span class="co"># #         Gene Ontology Enrichment Analysis         # # </span></span>
<span id="cb5-3"><a href="setup.html#cb5-3" tabindex="-1"></a><span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-4"><a href="setup.html#cb5-4" tabindex="-1"></a> </span>
<span id="cb5-5"><a href="setup.html#cb5-5" tabindex="-1"></a>callClusterProfilerFunc <span class="ot">&lt;-</span> <span class="cf">function</span>(result.obj.list, OrgDb){ </span>
<span id="cb5-6"><a href="setup.html#cb5-6" tabindex="-1"></a>   </span>
<span id="cb5-7"><a href="setup.html#cb5-7" tabindex="-1"></a>  <span class="co"># # Extract filtered lists from list container </span></span>
<span id="cb5-8"><a href="setup.html#cb5-8" tabindex="-1"></a>  geneNames.variable <span class="ot">&lt;-</span> result.obj.list<span class="sc">$</span>variable_genes[,<span class="dv">1</span>] </span>
<span id="cb5-9"><a href="setup.html#cb5-9" tabindex="-1"></a>  geneNames.up <span class="ot">&lt;-</span> result.obj.list<span class="sc">$</span>upregulated_genes[,<span class="dv">1</span>] </span>
<span id="cb5-10"><a href="setup.html#cb5-10" tabindex="-1"></a>  geneNames.down <span class="ot">&lt;-</span> result.obj.list<span class="sc">$</span>downregulated_genes[,<span class="dv">1</span>] </span>
<span id="cb5-11"><a href="setup.html#cb5-11" tabindex="-1"></a>  geneList <span class="ot">&lt;-</span> result.obj.list<span class="sc">$</span>vectorized_all_DE </span>
<span id="cb5-12"><a href="setup.html#cb5-12" tabindex="-1"></a>  universe <span class="ot">&lt;-</span> result.obj.list<span class="sc">$</span>all_genes[,<span class="dv">1</span>] </span>
<span id="cb5-13"><a href="setup.html#cb5-13" tabindex="-1"></a>   </span>
<span id="cb5-14"><a href="setup.html#cb5-14" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-15"><a href="setup.html#cb5-15" tabindex="-1"></a>  <span class="co"># #                       Analysis of Upregulated Genes                   # # </span></span>
<span id="cb5-16"><a href="setup.html#cb5-16" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-17"><a href="setup.html#cb5-17" tabindex="-1"></a>  <span class="do">## RUN WITH BP ONTOLOGY </span></span>
<span id="cb5-18"><a href="setup.html#cb5-18" tabindex="-1"></a>  <span class="co"># # GO Classification # # </span></span>
<span id="cb5-19"><a href="setup.html#cb5-19" tabindex="-1"></a>  go_class_upreg <span class="ot">&lt;-</span> <span class="fu">groupGO</span>( </span>
<span id="cb5-20"><a href="setup.html#cb5-20" tabindex="-1"></a>      <span class="at">gene =</span> geneNames.up, </span>
<span id="cb5-21"><a href="setup.html#cb5-21" tabindex="-1"></a>      <span class="at">OrgDb =</span> OrgDb, </span>
<span id="cb5-22"><a href="setup.html#cb5-22" tabindex="-1"></a>      <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb5-23"><a href="setup.html#cb5-23" tabindex="-1"></a>      <span class="at">level =</span> <span class="dv">3</span>, </span>
<span id="cb5-24"><a href="setup.html#cb5-24" tabindex="-1"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-25"><a href="setup.html#cb5-25" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) </span>
<span id="cb5-26"><a href="setup.html#cb5-26" tabindex="-1"></a>   </span>
<span id="cb5-27"><a href="setup.html#cb5-27" tabindex="-1"></a>  <span class="co"># # GO Over-Representation Analysis # # </span></span>
<span id="cb5-28"><a href="setup.html#cb5-28" tabindex="-1"></a>  go_overrep_upreg <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>( </span>
<span id="cb5-29"><a href="setup.html#cb5-29" tabindex="-1"></a>      <span class="at">gene =</span> geneNames.up, </span>
<span id="cb5-30"><a href="setup.html#cb5-30" tabindex="-1"></a>      <span class="at">universe =</span> universe, </span>
<span id="cb5-31"><a href="setup.html#cb5-31" tabindex="-1"></a>      <span class="at">OrgDb =</span> OrgDb, </span>
<span id="cb5-32"><a href="setup.html#cb5-32" tabindex="-1"></a>      <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb5-33"><a href="setup.html#cb5-33" tabindex="-1"></a>      <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, </span>
<span id="cb5-34"><a href="setup.html#cb5-34" tabindex="-1"></a>      <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-35"><a href="setup.html#cb5-35" tabindex="-1"></a>      <span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-36"><a href="setup.html#cb5-36" tabindex="-1"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-37"><a href="setup.html#cb5-37" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) </span>
<span id="cb5-38"><a href="setup.html#cb5-38" tabindex="-1"></a>   </span>
<span id="cb5-39"><a href="setup.html#cb5-39" tabindex="-1"></a>  <span class="co"># # Reactome Pathway Over-Representation Analysis # #  </span></span>
<span id="cb5-40"><a href="setup.html#cb5-40" tabindex="-1"></a>  reactome_overrep_upreg <span class="ot">&lt;-</span> <span class="fu">enrichPathway</span>( </span>
<span id="cb5-41"><a href="setup.html#cb5-41" tabindex="-1"></a>      <span class="at">gene =</span> geneNames.up,  </span>
<span id="cb5-42"><a href="setup.html#cb5-42" tabindex="-1"></a>      <span class="at">universe =</span> universe, </span>
<span id="cb5-43"><a href="setup.html#cb5-43" tabindex="-1"></a>      <span class="at">organism =</span> <span class="st">&#39;celegans&#39;</span>, </span>
<span id="cb5-44"><a href="setup.html#cb5-44" tabindex="-1"></a>      <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,  </span>
<span id="cb5-45"><a href="setup.html#cb5-45" tabindex="-1"></a>      <span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-46"><a href="setup.html#cb5-46" tabindex="-1"></a>      <span class="at">minGSSize =</span> <span class="dv">20</span>, </span>
<span id="cb5-47"><a href="setup.html#cb5-47" tabindex="-1"></a>      <span class="at">maxGSSize =</span> <span class="dv">500</span>, </span>
<span id="cb5-48"><a href="setup.html#cb5-48" tabindex="-1"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-49"><a href="setup.html#cb5-49" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) </span>
<span id="cb5-50"><a href="setup.html#cb5-50" tabindex="-1"></a>   </span>
<span id="cb5-51"><a href="setup.html#cb5-51" tabindex="-1"></a>   </span>
<span id="cb5-52"><a href="setup.html#cb5-52" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-53"><a href="setup.html#cb5-53" tabindex="-1"></a>  <span class="co"># #                      Analysis of Downregulated Genes                  # # </span></span>
<span id="cb5-54"><a href="setup.html#cb5-54" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-55"><a href="setup.html#cb5-55" tabindex="-1"></a>   </span>
<span id="cb5-56"><a href="setup.html#cb5-56" tabindex="-1"></a>  <span class="co"># # GO Classification # # </span></span>
<span id="cb5-57"><a href="setup.html#cb5-57" tabindex="-1"></a>  go_class_downreg <span class="ot">&lt;-</span> <span class="fu">groupGO</span>( </span>
<span id="cb5-58"><a href="setup.html#cb5-58" tabindex="-1"></a>      <span class="at">gene =</span> geneNames.down, </span>
<span id="cb5-59"><a href="setup.html#cb5-59" tabindex="-1"></a>      <span class="at">OrgDb =</span> OrgDb, </span>
<span id="cb5-60"><a href="setup.html#cb5-60" tabindex="-1"></a>      <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb5-61"><a href="setup.html#cb5-61" tabindex="-1"></a>      <span class="at">level =</span> <span class="dv">3</span>, </span>
<span id="cb5-62"><a href="setup.html#cb5-62" tabindex="-1"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-63"><a href="setup.html#cb5-63" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) </span>
<span id="cb5-64"><a href="setup.html#cb5-64" tabindex="-1"></a>   </span>
<span id="cb5-65"><a href="setup.html#cb5-65" tabindex="-1"></a>  <span class="co"># # GO Over-Representation Analysis # # </span></span>
<span id="cb5-66"><a href="setup.html#cb5-66" tabindex="-1"></a>  go_overrep_downreg <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>( </span>
<span id="cb5-67"><a href="setup.html#cb5-67" tabindex="-1"></a>      <span class="at">gene =</span> geneNames.down, </span>
<span id="cb5-68"><a href="setup.html#cb5-68" tabindex="-1"></a>      <span class="at">universe =</span> universe, </span>
<span id="cb5-69"><a href="setup.html#cb5-69" tabindex="-1"></a>      <span class="at">OrgDb =</span> OrgDb, </span>
<span id="cb5-70"><a href="setup.html#cb5-70" tabindex="-1"></a>      <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb5-71"><a href="setup.html#cb5-71" tabindex="-1"></a>      <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, </span>
<span id="cb5-72"><a href="setup.html#cb5-72" tabindex="-1"></a>      <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-73"><a href="setup.html#cb5-73" tabindex="-1"></a>      <span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-74"><a href="setup.html#cb5-74" tabindex="-1"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-75"><a href="setup.html#cb5-75" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) </span>
<span id="cb5-76"><a href="setup.html#cb5-76" tabindex="-1"></a>   </span>
<span id="cb5-77"><a href="setup.html#cb5-77" tabindex="-1"></a>  <span class="co"># # Reactome Pathway Over-Representation Analysis # #  </span></span>
<span id="cb5-78"><a href="setup.html#cb5-78" tabindex="-1"></a>  reactome_overrep_downreg <span class="ot">&lt;-</span> <span class="fu">enrichPathway</span>( </span>
<span id="cb5-79"><a href="setup.html#cb5-79" tabindex="-1"></a>      <span class="at">gene =</span> geneNames.down,  </span>
<span id="cb5-80"><a href="setup.html#cb5-80" tabindex="-1"></a>      <span class="at">universe =</span> universe, </span>
<span id="cb5-81"><a href="setup.html#cb5-81" tabindex="-1"></a>      <span class="at">organism =</span> <span class="st">&#39;celegans&#39;</span>, </span>
<span id="cb5-82"><a href="setup.html#cb5-82" tabindex="-1"></a>      <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>,  </span>
<span id="cb5-83"><a href="setup.html#cb5-83" tabindex="-1"></a>      <span class="at">qvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-84"><a href="setup.html#cb5-84" tabindex="-1"></a>      <span class="at">minGSSize =</span> <span class="dv">20</span>, </span>
<span id="cb5-85"><a href="setup.html#cb5-85" tabindex="-1"></a>      <span class="at">maxGSSize =</span> <span class="dv">500</span>, </span>
<span id="cb5-86"><a href="setup.html#cb5-86" tabindex="-1"></a>      <span class="at">readable =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb5-87"><a href="setup.html#cb5-87" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">desc</span>(Count)) </span>
<span id="cb5-88"><a href="setup.html#cb5-88" tabindex="-1"></a>   </span>
<span id="cb5-89"><a href="setup.html#cb5-89" tabindex="-1"></a>   </span>
<span id="cb5-90"><a href="setup.html#cb5-90" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-91"><a href="setup.html#cb5-91" tabindex="-1"></a>  <span class="co"># #                    GO Gene Set Enrichment Analysis                    # # </span></span>
<span id="cb5-92"><a href="setup.html#cb5-92" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-93"><a href="setup.html#cb5-93" tabindex="-1"></a>  go_gene_set_enrichment <span class="ot">&lt;-</span> <span class="fu">gseGO</span>( </span>
<span id="cb5-94"><a href="setup.html#cb5-94" tabindex="-1"></a>    <span class="at">geneList =</span> geneList, </span>
<span id="cb5-95"><a href="setup.html#cb5-95" tabindex="-1"></a>        <span class="at">OrgDb =</span> OrgDb, </span>
<span id="cb5-96"><a href="setup.html#cb5-96" tabindex="-1"></a>        <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>, </span>
<span id="cb5-97"><a href="setup.html#cb5-97" tabindex="-1"></a>        <span class="at">minGSSize =</span> <span class="dv">20</span>, </span>
<span id="cb5-98"><a href="setup.html#cb5-98" tabindex="-1"></a>        <span class="at">maxGSSize =</span> <span class="dv">500</span>, </span>
<span id="cb5-99"><a href="setup.html#cb5-99" tabindex="-1"></a>        <span class="at">pvalueCutoff =</span> <span class="fl">0.1</span>, </span>
<span id="cb5-100"><a href="setup.html#cb5-100" tabindex="-1"></a>        <span class="at">verbose =</span> <span class="cn">FALSE</span> </span>
<span id="cb5-101"><a href="setup.html#cb5-101" tabindex="-1"></a>        ) </span>
<span id="cb5-102"><a href="setup.html#cb5-102" tabindex="-1"></a>   </span>
<span id="cb5-103"><a href="setup.html#cb5-103" tabindex="-1"></a>  <span class="co"># change gene IDs from ENTREZID to symbol </span></span>
<span id="cb5-104"><a href="setup.html#cb5-104" tabindex="-1"></a>  go_gene_set_enrichment <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(go_gene_set_enrichment, <span class="at">OrgDb =</span> OrgDb, <span class="at">keyType=</span><span class="st">&quot;ENTREZID&quot;</span>) </span>
<span id="cb5-105"><a href="setup.html#cb5-105" tabindex="-1"></a>   </span>
<span id="cb5-106"><a href="setup.html#cb5-106" tabindex="-1"></a>   </span>
<span id="cb5-107"><a href="setup.html#cb5-107" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-108"><a href="setup.html#cb5-108" tabindex="-1"></a>  <span class="co"># #                       Reactome Enrichment Analysis                    # # </span></span>
<span id="cb5-109"><a href="setup.html#cb5-109" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-110"><a href="setup.html#cb5-110" tabindex="-1"></a> </span>
<span id="cb5-111"><a href="setup.html#cb5-111" tabindex="-1"></a>  reactome_gene_set_enrichment <span class="ot">&lt;-</span> <span class="fu">gsePathway</span>( </span>
<span id="cb5-112"><a href="setup.html#cb5-112" tabindex="-1"></a>    <span class="at">geneList =</span> geneList,  </span>
<span id="cb5-113"><a href="setup.html#cb5-113" tabindex="-1"></a>    <span class="at">organism =</span> <span class="st">&#39;celegans&#39;</span>, </span>
<span id="cb5-114"><a href="setup.html#cb5-114" tabindex="-1"></a>    <span class="at">pvalueCutoff =</span> <span class="fl">0.2</span>, </span>
<span id="cb5-115"><a href="setup.html#cb5-115" tabindex="-1"></a>    <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>,  </span>
<span id="cb5-116"><a href="setup.html#cb5-116" tabindex="-1"></a>    <span class="at">minGSSize =</span> <span class="dv">20</span>, </span>
<span id="cb5-117"><a href="setup.html#cb5-117" tabindex="-1"></a>    <span class="at">maxGSSize =</span> <span class="dv">500</span>, </span>
<span id="cb5-118"><a href="setup.html#cb5-118" tabindex="-1"></a>    <span class="at">verbose =</span> <span class="cn">FALSE</span> </span>
<span id="cb5-119"><a href="setup.html#cb5-119" tabindex="-1"></a>    ) </span>
<span id="cb5-120"><a href="setup.html#cb5-120" tabindex="-1"></a>   </span>
<span id="cb5-121"><a href="setup.html#cb5-121" tabindex="-1"></a>  <span class="co"># change Reactome GSE gene IDs from ENTREZID to symbol </span></span>
<span id="cb5-122"><a href="setup.html#cb5-122" tabindex="-1"></a>  reactome_gene_set_enrichment <span class="ot">&lt;-</span> <span class="fu">setReadable</span>(reactome_gene_set_enrichment, <span class="at">OrgDb =</span> OrgDb, <span class="at">keyType=</span><span class="st">&quot;ENTREZID&quot;</span>) </span>
<span id="cb5-123"><a href="setup.html#cb5-123" tabindex="-1"></a>   </span>
<span id="cb5-124"><a href="setup.html#cb5-124" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-125"><a href="setup.html#cb5-125" tabindex="-1"></a>  <span class="co"># #                             Return Results                            # # </span></span>
<span id="cb5-126"><a href="setup.html#cb5-126" tabindex="-1"></a>  <span class="co"># # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # </span></span>
<span id="cb5-127"><a href="setup.html#cb5-127" tabindex="-1"></a>   </span>
<span id="cb5-128"><a href="setup.html#cb5-128" tabindex="-1"></a>  go_results <span class="ot">&lt;-</span> <span class="fu">list</span>( </span>
<span id="cb5-129"><a href="setup.html#cb5-129" tabindex="-1"></a>    <span class="fu">list</span>(go_class_upreg), </span>
<span id="cb5-130"><a href="setup.html#cb5-130" tabindex="-1"></a>    <span class="fu">list</span>(go_overrep_upreg),  </span>
<span id="cb5-131"><a href="setup.html#cb5-131" tabindex="-1"></a>    <span class="fu">list</span>(reactome_overrep_upreg), </span>
<span id="cb5-132"><a href="setup.html#cb5-132" tabindex="-1"></a>    <span class="fu">list</span>(go_class_downreg), </span>
<span id="cb5-133"><a href="setup.html#cb5-133" tabindex="-1"></a>    <span class="fu">list</span>(go_overrep_downreg),  </span>
<span id="cb5-134"><a href="setup.html#cb5-134" tabindex="-1"></a>    <span class="fu">list</span>(reactome_overrep_downreg), </span>
<span id="cb5-135"><a href="setup.html#cb5-135" tabindex="-1"></a>    <span class="fu">list</span>(go_gene_set_enrichment), </span>
<span id="cb5-136"><a href="setup.html#cb5-136" tabindex="-1"></a>    <span class="fu">list</span>(reactome_gene_set_enrichment) </span>
<span id="cb5-137"><a href="setup.html#cb5-137" tabindex="-1"></a>    ) </span>
<span id="cb5-138"><a href="setup.html#cb5-138" tabindex="-1"></a>   </span>
<span id="cb5-139"><a href="setup.html#cb5-139" tabindex="-1"></a>  <span class="fu">names</span>(go_results) <span class="ot">&lt;-</span> <span class="fu">c</span>( </span>
<span id="cb5-140"><a href="setup.html#cb5-140" tabindex="-1"></a>    <span class="st">&quot;go_class_upreg&quot;</span>,  </span>
<span id="cb5-141"><a href="setup.html#cb5-141" tabindex="-1"></a>    <span class="st">&quot;go_overrep_upreg&quot;</span>,  </span>
<span id="cb5-142"><a href="setup.html#cb5-142" tabindex="-1"></a>    <span class="st">&quot;react_overrep_upreg&quot;</span>, </span>
<span id="cb5-143"><a href="setup.html#cb5-143" tabindex="-1"></a>    <span class="st">&quot;go_class_downreg&quot;</span>,  </span>
<span id="cb5-144"><a href="setup.html#cb5-144" tabindex="-1"></a>    <span class="st">&quot;go_overrep_downreg&quot;</span>,  </span>
<span id="cb5-145"><a href="setup.html#cb5-145" tabindex="-1"></a>    <span class="st">&quot;react_overrep_downreg&quot;</span>, </span>
<span id="cb5-146"><a href="setup.html#cb5-146" tabindex="-1"></a>    <span class="st">&quot;go_gse&quot;</span>, </span>
<span id="cb5-147"><a href="setup.html#cb5-147" tabindex="-1"></a>    <span class="st">&quot;react_gse&quot;</span> </span>
<span id="cb5-148"><a href="setup.html#cb5-148" tabindex="-1"></a>    ) </span>
<span id="cb5-149"><a href="setup.html#cb5-149" tabindex="-1"></a>   </span>
<span id="cb5-150"><a href="setup.html#cb5-150" tabindex="-1"></a>  <span class="fu">return</span>(go_results) </span>
<span id="cb5-151"><a href="setup.html#cb5-151" tabindex="-1"></a>} </span></code></pre></div>
</div>
</div>
<div id="read-in-data" class="section level2 hasAnchor">
<h2>Read in Data<a href="setup.html#read-in-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="setup.html#cb6-1" tabindex="-1"></a><span class="co"># Lysate</span></span>
<span id="cb6-2"><a href="setup.html#cb6-2" tabindex="-1"></a>lys <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/Documents/research_labs/mdibl_fall_2025/updike_lotr1_go_analysis/intermediate_data/deseq2_dds_objects/lotr1_isolatedStrain_Lysate_BatchCor_dds.rds&quot;</span>)</span>
<span id="cb6-3"><a href="setup.html#cb6-3" tabindex="-1"></a></span>
<span id="cb6-4"><a href="setup.html#cb6-4" tabindex="-1"></a><span class="co"># Polysome</span></span>
<span id="cb6-5"><a href="setup.html#cb6-5" tabindex="-1"></a>ply <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/Documents/research_labs/mdibl_fall_2025/updike_lotr1_go_analysis/intermediate_data/deseq2_dds_objects/lotr1_isolatedStrain_Polysome_BatchCor_dds.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="lotr-1-mutants" class="section level2 hasAnchor">
<h2>LOTR-1 Mutants<a href="setup.html#lotr-1-mutants" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><img src="../../mdibl_fall_2025/updike_lotr1_go_analysis/figures/lotr-1_mutants.png" width="618" /></p>
</div>
<div id="get-deseq2-results" class="section level2 hasAnchor">
<h2>Get DESeq2 Results<a href="setup.html#get-deseq2-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="setup.html#cb7-1" tabindex="-1"></a><span class="co"># Full KO vs WT</span></span>
<span id="cb7-2"><a href="setup.html#cb7-2" tabindex="-1"></a>full_vs_wt.lys <span class="ot">&lt;-</span> <span class="fu">results</span>(lys, <span class="at">name =</span> <span class="st">&quot;strain_DUP207_vs_DUP167&quot;</span>)</span>
<span id="cb7-3"><a href="setup.html#cb7-3" tabindex="-1"></a>full_vs_wt.ply <span class="ot">&lt;-</span> <span class="fu">results</span>(ply, <span class="at">name =</span> <span class="st">&quot;strain_DUP207_vs_DUP167&quot;</span>)</span>
<span id="cb7-4"><a href="setup.html#cb7-4" tabindex="-1"></a></span>
<span id="cb7-5"><a href="setup.html#cb7-5" tabindex="-1"></a><span class="co"># Tudor KO vs WT </span></span>
<span id="cb7-6"><a href="setup.html#cb7-6" tabindex="-1"></a>tudor_vs_wt.lys <span class="ot">&lt;-</span> <span class="fu">results</span>(lys, <span class="at">name =</span> <span class="st">&quot;strain_DUP185_vs_DUP167&quot;</span>)</span>
<span id="cb7-7"><a href="setup.html#cb7-7" tabindex="-1"></a>tudor_vs_wt.ply <span class="ot">&lt;-</span> <span class="fu">results</span>(ply, <span class="at">name =</span> <span class="st">&quot;strain_DUP185_vs_DUP167&quot;</span>)</span>
<span id="cb7-8"><a href="setup.html#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="setup.html#cb7-9" tabindex="-1"></a><span class="co"># LOTUS KO vs WT</span></span>
<span id="cb7-10"><a href="setup.html#cb7-10" tabindex="-1"></a>lotus_vs_wt.lys <span class="ot">&lt;-</span> <span class="fu">results</span>(lys, <span class="at">name =</span> <span class="st">&quot;strain_DUP217_vs_DUP167&quot;</span>)</span>
<span id="cb7-11"><a href="setup.html#cb7-11" tabindex="-1"></a>lotus_vs_wt.ply <span class="ot">&lt;-</span> <span class="fu">results</span>(ply, <span class="at">name =</span> <span class="st">&quot;strain_DUP217_vs_DUP167&quot;</span>)</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="full-lotr-1-ko-vs-wt-lysate.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/MasonMatich/bulk_RNA_seq_workflow/edit/main/GO_enrichment_analysis.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": "https://github.com/MasonMatich/bulk_RNA_seq_workflow/blob/main/GO_enrichment_analysis.Rmd",
    "text": null
  },
  "download": null,
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
